python-avd/pyavd/schema/generate_docs/yamllinegen.py

```diff
@@ -0,0 +1,412 @@
+# Copyright (c) 2023 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from abc import ABC
+from textwrap import indent
+from typing import TYPE_CHECKING, Generator
+
+import yaml
+from pydantic import BaseModel
+
+from .utils import render_schema_field
+
+if TYPE_CHECKING:
+    from ..metaschema.meta_schema_model import AvdSchemaField
+
+LEGACY_OUTPUT = True
+
+
+class YamlLine(BaseModel):
+    """
+    Dataclass for one table row.
+    Content of line is yaml formatted so it should be rendered in a code block.
+    The field may need an mkdocs code block annotation link to show large default values.
+    If so the content of the annotation is put into .annotation.
+
+    Since annotation numbers must be tracked across all yaml lines, it is up to the caller
+    to give us to annotation number at rendering time.
+
+    The annotation link can be rendered with .render_annotation_link(annotation_number).
+    The annotation foot note can be rendered with .render_annotation(annotation_number).
+    """
+
+    line: str
+    annotation: str | None = None
+
+    def __str__(self):
+        return self.line
+
+    def render_annotation(self, annotation_number: int) -> str:
+        """
+        Returns markdown for annotation foot note, providing the contents of mkdocs code block annotation popup
+
+        Like below (including the leading blank line):
+
+        1. Default Value
+
+            ```yaml
+            default: "one"
+            values: 2
+            ```
+        """
+        if self.annotation is None:
+            return ""
+        annotation_codeblock = f"```yaml\n{self.annotation}```"
+        return f"\n{annotation_number}. Default Value\n\n{indent(annotation_codeblock, '    ')}\n"
```

Comment:
so I assume for now our only use case with annotations is `Default Value`? The docstring of the class says "The field may need an mkdocs code block annotation link to show large default values."

Reply:
Correct we only use this for large default values.